src/main/java/seedu/address/logic/parser/FindContactCommandParser.java

```diff
@@ -0,0 +1,143 @@
+                PREFIX_EMAIL,
+                PREFIX_TAG,
+                PREFIX_DATE_AND_TIME,
+                PREFIX_DESCRIPTION,
+                PREFIX_CONTACT,
+                PREFIX_OF
+        );
+
+        if (!argMultimap.getPreamble().isEmpty()) {
+            throw new ParseException(
+                    String.format(MESSAGE_INVALID_COMMAND_FORMAT,
+                            FindContactCommand.MESSAGE_USAGE)
+            );
+        }
+        Predicate<Person> personPredicate = person -> true;
+        if (!arePrefixesEmpty(argMultimap, PREFIX_DATE_AND_TIME, PREFIX_DESCRIPTION, PREFIX_CONTACT, PREFIX_OF)) {
+            throw new ParseException(
+                    String.format(MESSAGE_INVALID_PREFIX, FindContactCommand.MESSAGE_USAGE));
+        }
+        // if all fields are empty
+        if (arePrefixesEmpty(argMultimap, PREFIX_NAME, PREFIX_ADDRESS, PREFIX_EMAIL, PREFIX_TAG)) {
+            throw new ParseException(
+                    String.format(MESSAGE_INVALID_COMMAND_FORMAT, FindContactCommand.MESSAGE_USAGE));
+        }
+        if (arePrefixesPresent(argMultimap, PREFIX_NAME)) {
+            assert argMultimap.getValue(PREFIX_NAME).isPresent();
+            String nameKeyWord =
+                    argMultimap.getValue(PREFIX_NAME).get().trim().toLowerCase();
+            personPredicate =
+                    personPredicate.and(person -> person.getName().fullName
+                            .toLowerCase()
+                            .contains(nameKeyWord));
+        }
+        if (arePrefixesPresent(argMultimap, PREFIX_EMAIL)) {
+            assert argMultimap.getValue(PREFIX_EMAIL).isPresent();
+            String emailKeyWord =
+                    argMultimap.getValue(PREFIX_EMAIL).get().trim().toLowerCase();
+            personPredicate =
+                    personPredicate.and(person -> {
+                        if (person.getEmail().equals(Email.EMPTY_EMAIL)) {
+                            return false;
+                        } else {
+                            return person.getEmail().value
+                                    .toLowerCase()
+                                    .contains(emailKeyWord);
+                        }
+                    });
+        }
+        if (arePrefixesPresent(argMultimap, PREFIX_ADDRESS)) {
+            assert argMultimap.getValue(PREFIX_ADDRESS).isPresent();
+            String addressKeyWord =
+                    argMultimap.getValue(PREFIX_ADDRESS).get().trim().toLowerCase();
+            personPredicate =
+                    personPredicate.and(person -> {
+                        if (person.getAddress().equals(Address.EMPTY_ADDRESS)) {
+                            return false;
+                        } else {
+                            return person.getAddress().value
+                                    .toLowerCase()
+                                    .contains(addressKeyWord);
+                        }
+                    });
```

Comment:
May want to move the predicates elsewhere to SLAP harder. This method is quite long at about 90 lines, so shortening the predicate declaration would make it a lot more readable. Maybe a method that constructs a `Predicate<Person>` with the given keyword?